tracer/src/Datadog.Trace/Logging/DirectSubmission/Sink/PeriodicBatching/BatchingSink.cs

```diff
@@ -175,6 +177,8 @@ private async Task<CircuitStatus> FlushLogs()
                 var haveMultipleBatchesToSend = false;
                 do
                 {
+                    TelemetryFactory.Metrics.RecordGaugeDirectLogQueue(_queue.Count);
```

Comment:
Getting the Count of a ConcurrentQueue is costly on .net core. It's better to increment a counter when we dequeue the items in the while loop.

Reply:
Good to know, cheers, I'll add the fix in [this PR](https://github.com/DataDog/dd-trace-dotnet/pull/4291) to avoid another CI cycle

Reply:
Actually, it's not as simple as that 🤔 we can't guarantee that we actually try to send all the logs (because it depends on the API), so we don't necessarily dequeue them all. 

As this is running in a separate background thread, hopefully the cost won't be too much of an issue? 🤔

Reply:
actually, scratch all that, `queue` is our `BoundedConcurrentQueue<T>`, which exposes `Count`, and increments that independently of the queue, so I think we're good 🙂